Druid/DRBasicBlock.class.st

```diff
@@ -197,6 +197,24 @@ DRBasicBlock >> breakBy: instruction [
 		  self }
 ]
 
+{ #category : #asserting }
+DRBasicBlock >> canBeLinearizedBefore: aDRBasicBlock [
+
+	self = aDRBasicBlock ifTrue: [ ^ true ].
+
+	"Domination constraints"
+	(self isDominatedBy: aDRBasicBlock) ifTrue: [ ^ false ].
+
+	"Pops and Pushes are constraints for linearization"
+	controlFlowGraph stackInstructions do: [ :stackInstruction | "There is a stack instruction in my path ..."
```

Comment:
I wonder if we don't need to put this second part of the sorting in a separate (sub)class. This is only required to stage push/pop instructions of the abstract interpreter, which only happens in one of our backends.

Also, I wonder if this cannot just be rewritten a bit simpler as a topological sort with extra constraints.

Reply:
https://github.com/Alamvic/druid/issues/91